### PR TITLE
gpu: Update to HPC 22.3, reduced image size

### PR DESCRIPTION
**Pull request by kenhester**

### CODE REVIEW

File: docker/Dockerfile.nvidia [hunk 1]
```diff
@@ -16,41 +16,52 @@
...
+#   docker run --gpus all --rm -it devito:nvidia python examples/seismic/acoustic/acoustic_example.py
 #
 #   docker run --gpus all --rm -it -p 8888:8888 -p 8787:8787 -p 8786:8786 --device=/dev/infiniband/uverbs0 --device=/dev/infiniband/rdma_cm  devito:nvidia 
 #
 # to run in user context on a cluster with shared filesystem, you can add the correct user config as docker options e.g.:
 #   docker run --gpus all --rm -it -v `pwd`:`pwd` -w `pwd` -u $(id -u):$(id -g) devito:nvidia python examples/seismic/acoustic/acoustic_example.py
 #
 ##############################################################
-FROM python:3.8
+#FROM python:3.9
```
Comment (georgebisbas) [inline docker/Dockerfile.nvidia:36]: Are the comments to be removed?

**Review by georgebisbas — COMMENTED** [review]
Thanks!

File: docker/Dockerfile.nvidia (hunk 1 above)
Comment (kenhester) [reply docker/Dockerfile.nvidia:36]: I was just try to illustrate that there was an alternate non-slim version of the base container. `#FROM python:3.9 ` could be removed, yes.

Comment (georgebisbas) [reply docker/Dockerfile.nvidia:36]: I think, the slimmer, the better right?

File: docker/Dockerfile.nvidia [hunk 2]
```diff
@@ -16,41 +16,52 @@
...
 RUN echo 'deb [trusted=yes] https://developer.download.nvidia.com/hpc-sdk/ubuntu/amd64 /' > /etc/apt/sources.list.d/nvhpc.list && \
     apt-get update -y && \
     apt-get install -y -q \
-        apt-utils curl\   
-        vim && \
+        apt-utils curl wget \   
     curl -sL https://deb.nodesource.com/setup_12.x | bash - && \
     apt-get install -y -q \
-        nodejs \
+#        nvhpc-22-3-cuda-multi \
```
Comment (georgebisbas) [inline docker/Dockerfile.nvidia:48]: same I guess

File: docker/Dockerfile.nvidia (hunk 1 above)
Comment (kenhester) [reply docker/Dockerfile.nvidia:36]: That is the ask from some of our the users of the container.

**Comment by mloubout** [comment]
Could we make this automatic somehow with a cron workflow to avoid the effort and PRs. Like just check once a week if there is a new sdk and update the image

**Comment by FabioLuporini** [comment]
> Could we make this automatic somehow with a cron workflow to avoid the effort and PRs. Like just check once a week if there is a new sdk and update the image

That's a lovely idea -- tho that's probably on us ;)

File: docker/Dockerfile.nvidia (hunk 2 above)
Comment (mloubout) [reply docker/Dockerfile.nvidia:48]: Two previous cuda version and everythng that goes with it. Takes a lot of space usually yeah and not needed.

**Comment by FabioLuporini** [comment]
@mloubout @georgebisbas can you make any sense of the conflict with the existing Dockerfile.nvidia ?

**Comment by georgebisbas** [comment]
> @mloubout @georgebisbas can you make any sense of the conflict with the existing Dockerfile.nvidia ?

Can I see it somehow?

**Comment by georgebisbas** [comment]
> @mloubout @georgebisbas can you make any sense of the conflict with the existing Dockerfile.nvidia ?

I wonder whether this conflict comes from @kenhester 's fork?
@kenhester could you kindly double-check?

**Comment by mloubout** [comment]
Conflict is from the changes to dockerfile that were made in between yes since the COPY has slightly changed.

**Comment by mloubout** [comment]
Should be GTG

**Comment by FabioLuporini** [comment]
Thanks a lot @kenhester ! Merging, sorry it took a while, we have had a long list of PRs in the queue

**Pull request merged**